docs/reference/settings/ml-settings.asciidoc

```diff
@@ -28,17 +28,19 @@ file.
 [[general-ml-settings]]
 ==== General machine learning settings
 
-`node.ml`::
-deprecated:[7.9.0,"Use <<modules-node,node.roles>> instead."]
-Set to `true` (default) to identify the node as a _machine learning node_. +
+`node.roles: [ ml ]`::
+Set `node.roles` to contain `ml` to identify the node as a _{ml} node_. +
 +
-If set to `false` in `elasticsearch.yml`, the node cannot run jobs. If set to
-`true` but `xpack.ml.enabled` is set to `false`, the `node.ml` setting is
-ignored and the node cannot run jobs. If you want to run jobs, there must be at
-least one machine learning node in your cluster. +
+If a node does not have the `ml` role in `elasticsearch.yml`, or it has the role 
+but `xpack.ml.enabled` is set to `false`, the node cannot run jobs. If you want 
+to run jobs, there must be at least one {ml} node in your cluster. +
```

Comment:
Now explained in prev para. Suggest leaving as 

`If you want to run jobs, there must be at least one {ml} node in your cluster.`